app/controllers/settings/payments_controller.rb

```diff
@@ -1,17 +1,20 @@
 # frozen_string_literal: true
 
-class Settings::PaymentsController < Sellers::BaseController
+class Settings::PaymentsController < Settings::BaseController
   include ActionView::Helpers::SanitizeHelper
 
   before_action :authorize
 
   def show
     @title = "Settings"
-    @react_component_props = SettingsPresenter.new(pundit_user:).payments_props(remote_ip: request.remote_ip)
+
+    render inertia: "Settings/Payments", props: settings_presenter.payments_props(remote_ip: request.remote_ip)
   end
 
   def update
-    return render(json: { success: false, error_message: "You have to confirm your email address before you can do that." }) unless current_seller.email.present?
+    unless current_seller.email.present?
+      return re_render_with_error("You have to confirm your email address before you can do that.")
```

Comment:
We re-render the Inertia page with error props (error_message and error_code) instead of using redirect_to because the frontend component (PaymentsPage.tsx) reads these props to display an error banner, redirect_to only supports flash messages (not custom props), and redirect_to with Inertia triggers onSuccess rather than onError, so we can't handle validation errors that way. Re-rendering with a 422 status preserves form state and allows the component to display inline error messages based on the error props.